clang/test/SemaSYCL/parallel_for_wrapper_attr.cpp

```diff
@@ -0,0 +1,50 @@
+// RUN: %clang_cc1 %s -fsyntax-only -ast-dump -fsycl -fsycl-is-device -triple spir64 | FileCheck %s
+
+template <typename Name, typename Type>
+[[clang::sycl_kernel]] void __my_kernel__(Type bar) {
+  bar();
+}
+
+template <typename Name, typename Type> void parallel_for(Type lambda) {
+  __my_kernel__<Name>(lambda);
+}
+
+template <typename T> class Fobj {
+public:
+  Fobj() {}
+  void operator()() const {
+    auto L0 = []() [[intel::reqd_sub_group_size(4)]]{};
+    L0();
+  }
+};
+
+void invoke() {
+  Fobj<int> fobj1;
```

Comment:
Can you also explain what is happening in this test? I don't fully understand the 2 cases.